upstairs/src/downstairs.rs

```diff
@@ -523,27 +515,19 @@ impl Downstairs {
         // Specifically, we want to skip jobs if the only path back online for
         // that client goes through live-repair; if that client can come back
         // through replay, then the jobs must remain live.
-        let new_state = self.clients[client_id].state();
-        if matches!(prev_state, DsState::LiveRepair | DsState::Active)
```

Comment:
It was impossible to go from `Active` → `Faulted`; in the previous implementation of `on_missing`, `Active` always became `Offline`.  Therefore, this branch only fired if the incoming state is `LiveRepair`.